### DISCUSSION TITLE
Предложение: использовать выпуск версий github

### DISCUSSION THREAD

**Discussion by ergashev** [qzeleza/kvas#34] [Ideas]
Приветствую!
Хотелось бы иметь возможность автоматизировать загрузку новой версии кваса.
Сейчас для того что-бы получить ссылку на последнюю версию кваса нужно знать имя файла, но т.к. оно не уникально это усложняет автоматическое получение новой версии.
У гитхаба есть возможность сделать уникальную ссылку на последнюю версию файла: https://docs.github.com/ru/repositories/releasing-projects-on-github/linking-to-releases
Но, для этого нужно использовать функционал выпуска версий.
Если имеется возможность, рассмотрите, пожалуйста, использование функционала выпуска релизов на гитхаб.

**Comment by ergashev** [comment]
@qzeleza

**Comment by qzeleza** [comment]
Здравствуйте, 
Готов рассмотреть внедрение данного функционала, в случае, если Вы сможете помочь мне освоить его. 
Сейчас пока нет понимания для чего автоматизировать данный процесс, а во вторых я не совсем разобрался, как работает данный механизм. Как-то попытался собрать здесь, но меня поставили в очередь обработки, которая длилась почти 3 дня и я так и не дождался, понял, что самый оптимальный вариант - это сборка своими средствами. 
Но возможно я делал что-то не так, допускаю. Потому, если у Вас есть понимание данного процесса и опыт в этом направлении, которым Вы готовы поделиться, то давайте пробывать.

**Comment by ergashev** [comment]
Действительно, небольшие проекты быстрее собирать самостоятельно, по этому на гитхабе есть возможность выпускать версии уже заранее собранных приложений. По сути, это просто публикация релиза.
Для того что бы опубликовать версию, нужно со страницы [релизов](https://github.com/qzeleza/kvas/releases) перейти в [Draft a new release](https://github.com/qzeleza/kvas/releases/new) и заполнить:
- тег (обычно это номер версии, например: v1.1.2)
- выбрать нужную ветку (обычно это dev или master ветка, в вашем случае main)
- название
- описание
- прикрепить заранее собранный файл

И этого достаточно для того что бы опубликовать версию.

Перед описанием процесса я сделал форк Вашего репозитория, что бы проверить что ничего кардинально не поменялось. Все работает, для примера:
[https://github.com/ergashev/kvas/releases](https://github.com/ergashev/kvas/releases)

**Reply by qzeleza** [reply]
Доброго утра,
Опубликовал. Только поясните мне пожалуйста цель этой публикации? Чем плохо зайти в папку и скопировать от туда?

**Reply by ergashev** [reply]
Доброе утро! Благодарю! Копировать не плохо, просто не совсем удобно это делать каждый раз вручную, работая только из консоли. А так теперь скачивание можно уже и автоматизировать и по прямой ссылке и через API GitHub

**Comment by kilia07** [comment]
Доброй ночи! 
А еще лучше или в довесок добавить обновление kvas из entware в автомате.
Что-то типа:
kvas upgrade

**Reply by qzeleza** [reply]
Доброго утра
Идея интересная, спасибо - подумаю над реализацией.

**Reply by ergashev** [reply]
Соглашусь, обновлять из репозитория entware куда  удобнее, но это сложнее в реализации.
